packages/core/src/analytics.ts

```diff
@@ -568,7 +568,9 @@ export class SegmentClient {
       },
     });
 
-    this.store.dispatch(this.actions.userInfo.setUserId({ userId }));
+    if (userId) {
```

Comment:
Let's put a strict undefined check here:
```suggestion
    if (userId !== undefined) {
```